traffic/street_lights_client.py

```diff
@@ -1,44 +1,108 @@
+###TODO TEST PUBBLISHED MESSAGES
+
 # MQTT Client demo
 # Continuously monitor two different MQTT topics for data,
 # check if the received data matches two predefined 'commands'
- 
 import paho.mqtt.client as mqtt
+import json
 from grovepi import *
 from time import sleep
 from configparser import ConfigParser 
-
+##read values from config ini   
 file = 'config.ini'
 config = ConfigParser()
 config.read(file)
+
+
+##topic constructor
+def topic_constructor(protocol, service_api_key, sensor_id):
+   return ("/" + protocol + "/" + service_api_key + "/" + sensor_id)
+
+def attr_topic(topic):
+    return (topic + "/attrs")
+
+def cmd_topic(topic):
+    return (topic + "/cmd")
+
+def ack_topic(cmd_topic):
+    return (cmd_topic + "exe")
+
+
 #config connection
 broker_address=str(config['connection']['broker_address'])
 port = int(config['connection']['port'])
 username = str(config['connection']['username'])
 password = str(config['connection']['password'])
-#######street1 config
-#street led 1 topic
-vcln_topic=str(config['Street 1']['vcln_topic'])
-sl1_topic = str(config['Street 1']['lt'])
-vcln_pin = int(config['Street 1']['vcln_pin'])
-#led pins
-l1_pin = int(config['Street 1']['l1_pin'])
-l2_pin = int(config['Street 1']['l2_pin'])
-l3_pin = int(config['Street 1']['l3_pin'])
-l4_pin = int(config['Street 1']['l4_pin'])
-#######street2 config
-#street led 1 topic
-os_topic = str(config['Street 2']['os_topic'])
-sl2_topic= str(config['Street 2']['lt'])
-os_pin = int(config['Street 2']['os_pin'])
-l5_pin = int(config['Street 2']['l1_pin'])
-l6_pin = int(config['Street 2']['l2_pin'])
-l7_pin = int(config['Street 2']['l3_pin'])
-l8_pin = int(config['Street 2']['l4_pin'])
-
-##flag to see when connection active
-connected = False
-##keep car count
-car_counter = 0
+##topic base config
+protocol = "json" #str(config['connection']['protocol'])
+service_api_key = "test_key"#str(config['Street 2']['service_api_key'])
```

Comment:
this should be passed as variable?

Reply:
yes the code to pass it as a variable is the commented one, i left it like that just for speed up coding since there isn't a definitive structure of the config.ini file

Reply:
let's fix it, or let's open a ticket to remind about it.

Reply:
car counter removed
config.ini structure defined and passed variables